### DISCUSSION TITLE
How to map username to family_name+ given_name ?

### DISCUSSION THREAD

**Discussion by AlwaysNoobCoder** [keycloak/keycloak#30844] [Q&A]
Hi, I have a keycloak instance and a openid client .

our user model is something like below:

username: 06052 
family_name: 张
given_name: 三

for the openid client, I want to map `username -> family_name + given_name`

results:
username: 张三

how can I do this ?

**Comment by lethargosapatheia** [comment]
Funny, I have the exactly same issue. I've no idea what the correct syntax is. So if I go to Identity Providers - Google (created by me) - Mappers, I select "Attribute Importer" and then input "given_name" in "Social profile JSON Field Path" and select "User Attribute Name", it works in that the username will be just the forename (as given by Google through the "given_name" attribute).

Combining "given_name" and "family_name" doesn't seem to work at first glance. I'm not sure how to get the syntax right or if keycloak supports this at all.

**Comment by lethargosapatheia** [comment]
Ok, I've got it. In order to make it work, you need to choose `Username Template Importer` in the "Mapper Type".
Then in the "Template" field you have to add this:
```
${CLAIM.given_name}.${CLAIM.family_name}
```
And this works for Google.
The target can be "LOCAL". At least this works for me.

**Comment by lamoboos223** [comment]
where do you want that fabricated `username` to be returned? in the `access_token`? if yes, you should be able to do this by creating new field and name it `username` and the value is the combination of `family_name` + `given_name` from `user attributes`


@AlwaysNoobCoder
